blog/2024-06-14-using-table-component.md

```diff
@@ -0,0 +1,486 @@
+Lastly the `State.load_entries` event handler is called when the `rx.table` is rendered using the `on_mount` event trigger.
+
+```python
+class State(rx.State):
+    """The app state."""
+
+    users: list[Customer] = []
+    current_user: Customer = Customer()
+
+    def load_entries(self) -> list[Customer]:
+        """Get all users from the database."""
+        with rx.session() as session:
+            query = select(Customer)
+            self.users = session.exec(query).all()
+
+
+def show_customer(user: Customer):
+    """Show a customer in a table row."""
+    return rx.table.row(
+        rx.table.cell(user.name),
+        rx.table.cell(user.email),
+        rx.table.cell(user.phone),
+        rx.table.cell(user.address),
+        rx.table.cell(user.payments),
+        rx.table.cell(user.date),
+        rx.table.cell(
+            rx.match(
+                user.status,
+                ("Delivered", status_badge("Delivered")),
+                ("Pending", status_badge("Pending")),
+                ("Cancelled", status_badge("Cancelled")),
+                status_badge("Pending")
+            )
+        ),
+    )
+
+def main_table():
+    return rx.table.root(
+            rx.table.header(
+                rx.table.row(
+                    _header_cell("Name", "user"),
+                    _header_cell("Email", "mail"),
+                    _header_cell("Phone", "phone"),
+                    _header_cell("Address", "home"),
+                    _header_cell("Payments", "dollar-sign"),
+                    _header_cell("Date", "calendar"),
+                    _header_cell("Status", "truck"),
+                    _header_cell("Actions", "cog"),
+                ),
+            ),
+            rx.table.body(rx.foreach(State.users, show_customer)),
+            on_mount=State.load_entries,
+        )
+
+
+app = rx.App(
+    theme=rx.theme(appearance="light", has_background=True, radius="large", accent_color="grass"),
+    )
+app.add_page(main_table)
+```
+
+
+
+
+
+## Adding a New Customer 
+
+Next let's explore how to add a user to our database and therefore to our table. We have a function `add_customer_button`, which is a UI element that is rendered in the `main_table` function defined in the section above. 
+
+In this `add_customer_button` function there is an `rx.dialog` which opens when the `rx.button` "Add New Customer" is pressed. When the dialog is opened there is an `rx.form.root` with several `form_field` functions called inside. The `form_field` function returns an `rx.form.field`. It takes in several arguments, but most importantly takes in a `name` argument. The `name` prop is needed to submit with its owning form as part of a `name/value` pair. The content typed into the `rx.input` is submitted as the `value` in that `name/value` pair when the `rx.form` is submitted. The same is done for the `rx.radio` component, where the `name` is set and the radio item selected is passed as the `value` when the `rx.form` is submitted.
```

Comment:
```suggestion
In this `add_customer_button` function there is an `rx.dialog` which opens when the `rx.button` "Add New Customer" is pressed. When the dialog is opened there is an `rx.form.root` with several `form_field` functions called inside. The `form_field` function returns an `rx.form.field`. It takes in several arguments, but most importantly takes in a `name` argument. The `name` prop is needed to submit with its owning form as part of a name/value pair. The content typed into the `rx.input` is submitted as the `value` in that name/value pair when the `rx.form` is submitted. The same is done for the `rx.radio` component, where the `name` is set and the radio item selected is passed as the `value` when the `rx.form` is submitted.
```

same as above, reserve the code ticks for something that you literally see in the code

Reply:
fixed